subsys/net/lib/openthread/platform/radio.c

```diff
@@ -1019,23 +1019,113 @@ uint8_t otPlatRadioGetCslAccuracy(otInstance *aInstance)
 }
 
 #if defined(CONFIG_OPENTHREAD_LINK_METRICS)
+/**
+ * Header IE format - IEEE Std. 802.15.4-2015, 7.4.2.1 && 7.4.2.2
+ *
+ * +---------------------------------+----------------------+
+ * | Length    | Element ID | Type=0 |      Vendor OUI      |
+ * +-----------+------------+--------+----------------------+
+ * |           Bytes: 0-1            |          2-4         |
+ * +-----------+---------------------+----------------------+
+ * | Bits: 0-6 |    7-14    |   15   | IE_VENDOR_THREAD_OUI |
+ * +-----------+------------+--------+----------------------|
+ *
+ * Thread v1.2.1 Spec., 4.11.3.4.4.6
+ * +---------------------------------+-------------------+------------------+
+ * |                  Vendor Specific Information                           |
+ * +---------------------------------+-------------------+------------------+
+ * |                5                |         6         |   7 (optional)   |
+ * +---------------------------------+-------------------+------------------+
+ * | IE_VENDOR_THREAD_ACK_PROBING_ID | LINK_METRIC_TOKEN | LINK_METRIC_TOKEN|
+ * |---------------------------------|-------------------|------------------|
+ */
+static uint8_t set_vendor_ie_header(bool lqi, bool link_margin, bool rssi, uint8_t *ie_header)
```

Comment:
Rename function to something more specific to link metrics? Vendor IE header is a generic IEEE 802.15.4 term.